src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -154,7 +155,13 @@ public function load(array $configs, ContainerBuilder $container): void
                 ));
         }
 
-        $container->setParameter('security.authentication.hide_user_not_found', $config['hide_user_not_found']);
+        if (true === $config['hide_user_not_found']) {
```

Comment:
I think the BC layer should instead of handled in the MainConfiguration class, by making `beforeNormalization` set the `expose_security_errors` value when it is not set while `hide_user_not_found` is set (keeping `hide_user_not_found` as a known property without default value, just for the deprecation reporting part).
This way, things will keep working when merging multiple configurations together, where some of them are migrated but not all. The current implementation makes any `hide_user_not_found` win, even if it is specified in the first config and `expose_security_errors` is defined in a later config that should win over it.

Reply:
Moved the BC handling to the `MainConfiguration` as proposed.